### DISCUSSION TITLE
Do I need an frontend proxy for uWebSockets?

### DISCUSSION THREAD

**Discussion by uasan** [uNetworking/uWebSockets.js#188] [Q&A] [answered]
Hi, we use Nginx to balance and cache endpoints responses.
I am considering the possibility of removing proxies and implementing all the caching and balancing logic on uWebSockets.js.

Risks and problems that I see:

1. HTTP/2 not implemented in uWebSockets
2. Slow client connections, take a lot of memory and time.
3. HTTP vulnerabilities, I know less about them than the Nginx developers

Maybe you have experience or good advice?
Thanks.

**Comment by ghost** [comment, marked as answer]
I don't know, you'll have to test
